packages/core/src/wot-impl.ts

```diff
@@ -39,8 +41,19 @@ export default class WoTImpl {
         throw new Error("not implemented");
     }
 
+    /** @inheritDoc */
     async requestThingDescription(url: string): Promise<WoT.ThingDescription> {
-        throw new Error("not implemented");
+        const uriScheme = new URL(url).protocol.split(":")[0];
+        const client = this.srv.getClientFor(uriScheme);
+        const content = await client.requestThingDescription(url);
+
+        const value = ContentManager.contentToValue({ type: content.type, body: await content.toBuffer() }, TDSchema);
+
+        if (value instanceof Object) {
+            return value as WoT.ThingDescription;
+        }
```

Comment:
I got rid of the TODO regarding validation now by passing the `TDSchema` to the `contentToValue` method alongside the TD content. I hope that makes sense – more validation could probably be added later.

Reply:
As I recall `contentToValue` method does not do the validation. It uses the data schema only as a hint when converting. The proper way to handle this should use AJV here. But we have to think this correctly, given #1062 .

Reply:
In 5aeb45dde8507cb205e5f5c406e0b1f7528a15f1, I now incorporated the validation code used by @danielpeintner in #1169 instead